watertap/unit_models/translator_adm1_asm1.py

```diff
@@ -0,0 +1,463 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Modified translator block.
+
+This is copied from the Generic template for a translator block.
+
+Assumptions:
+     * Steady-state only
+
+Model formulated from:
+
+Copp J. and Jeppsson, U., Rosen, C., 2006.
+Towards an ASM1 - ADM1 State Variable Interface for Plant-Wide Wastewater Treatment Modeling.
+ Proceedings of the Water Environment Federation, 2003, pp 498-510.
+"""
+
+# Import Pyomo libraries
+from pyomo.common.config import ConfigBlock, ConfigValue, In, Bool
+
+# Import IDAES cores
+from idaes.core import declare_process_block_class, UnitModelBlockData
+from idaes.core.util.config import (
+    is_physical_parameter_block,
+    is_reaction_parameter_block,
+)
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core.solvers import get_solver
+import idaes.logger as idaeslog
+
+from pyomo.environ import (
+    Constraint,
+    Param,
+    units as pyunits,
+    check_optimal_termination,
+    Set,
+)
+
+__author__ = "Alejandro Garciadiego, Andrew Lee"
+
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("Translator_ADM1_ASM1")
+class TranslatorData(UnitModelBlockData):
+    """
+    Standard Translator Block Class
+    """
+
+    CONFIG = ConfigBlock()
```

Comment:
If we inherited from the Translator block class, I think we wouldn't need to include _most_ of these config options--only the newly added ones.